src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,109 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import React from 'react';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const checkForEmptyField = () => {
```

Comment:
you can put the result of the expression (validation) into variable directly since it will be re-calculated on each re-render (state change):
```suggestion
  const areFieldsValid = title && imgUrl && imdbUrl && imdbId;
```